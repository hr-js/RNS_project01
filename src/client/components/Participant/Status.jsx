// @flow
import * as React from "react";
import styles from "./style.css";

type StatusType = {
  isEntry: boolean
};

function Status(props: StatusType): React.Node {
  const { isEntry } = props;
  const { columnStatus, entry, exit } = styles;

  const className = isEntry
    ? `${columnStatus} ${entry}`
    : `${columnStatus} ${exit}`;
  const value = isEntry ? "入" : "退";

  return <div className={className}>{value}</div>;
}
export default Status;
