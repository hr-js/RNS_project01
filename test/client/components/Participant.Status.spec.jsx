import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Status from "@components/Participant/Status";

describe("Participant.Status.jsxjのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンタリング", () => {
      const tree = renderer.create(<Status isEntry />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    describe("属性にisEntryを指定した時", () => {
      it("共通のcssクラスの他にentryをもち、入と表示される", () => {
        const status = shallow(<Status isEntry />);
        expect(status.hasClass("columnStatus")).toBeTruthy();
        expect(status.hasClass("entry")).toBeTruthy();
        expect(status.text("入")).toBeTruthy();
      });
    });

    describe("属性にisEntryを指定しない時", () => {
      it("共通のcssクラスの他にexitをもち、退と表示される", () => {
        const status = shallow(<Status />);
        expect(status.hasClass("columnStatus")).toBeTruthy();
        expect(status.hasClass("exit")).toBeTruthy();
        expect(status.text("退")).toBeTruthy();
      });
    });
  });
});
