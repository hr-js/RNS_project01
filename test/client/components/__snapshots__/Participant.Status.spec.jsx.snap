// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant.Status.jsxjのテスト スナップショット 正しいレンタリング 1`] = `
<div
  className="columnStatus entry"
>
  入
</div>
`;
